processing/src/main/java/io/druid/query/metadata/SegmentMetadataQueryConfig.java

```diff
@@ -33,8 +35,23 @@
   @JsonProperty
   private Period defaultHistory = ISO_FORMATTER.parsePeriod(DEFAULT_PERIOD_STRING);
 
+  @JsonProperty
+  private EnumSet<SegmentMetadataQuery.AnalysisType> defaultAnalysisType = SegmentMetadataQuery.DEFAULT_ANALYSIS_TYPES;
+
+  public SegmentMetadataQueryConfig(String period, EnumSet<SegmentMetadataQuery.AnalysisType> analysisType)
```

Comment:
This constructor is currently never used, It will work I guess, but I didn't write tests that required a `SegmentMetadataQueryConfig` with both defaultAnalysisType and period being set, as they are independent.
I'll remove it considering it's not being used.